program/rust/src/tests/test_upd_price_with_validator.rs

```diff
@@ -0,0 +1,380 @@
+    let mut price_account = price_setup.as_account_info();
+    price_account.is_signer = false;
+    PriceAccount::initialize(&price_account, PC_VERSION).unwrap();
+
+    {
+        let mut price_data = load_checked::<PriceAccount>(&price_account, PC_VERSION).unwrap();
+        price_data.num_ = 1;
+        price_data.comp_[0].pub_ = *funding_account.key;
+        price_data.flags.insert(PriceAccountFlags::ACCUMULATOR_V2);
+        price_data
+            .flags
+            .insert(PriceAccountFlags::MESSAGE_BUFFER_CLEARED);
+    }
+
+    let mut clock_setup = AccountSetup::new_clock();
+    let mut clock_account = clock_setup.as_account_info();
+    clock_account.is_signer = false;
+    clock_account.is_writable = false;
+
+    update_clock_slot(&mut clock_account, 1);
+
+    assert!(process_instruction(
+        &program_id,
+        &[
+            funding_account.clone(),
+            price_account.clone(),
+            clock_account.clone()
+        ],
+        &instruction_data
+    )
+    .is_ok());
+
+
+    {
+        let price_data = load_checked::<PriceAccount>(&price_account, PC_VERSION).unwrap();
+        assert_eq!(price_data.comp_[0].latest_.price_, 42);
+        assert_eq!(price_data.comp_[0].latest_.conf_, 2);
+        assert_eq!(price_data.comp_[0].latest_.pub_slot_, 1);
+        assert_eq!(price_data.comp_[0].latest_.status_, PC_STATUS_TRADING);
+        assert_eq!(price_data.valid_slot_, 0);
+        assert_eq!(price_data.agg_.pub_slot_, 0);
+        assert_eq!(price_data.agg_.price_, 0);
+        assert_eq!(price_data.agg_.status_, PC_STATUS_UNKNOWN);
+    }
+
+    // a publisher's component pub_slot_ has to be strictly increasing -- get rejected
+    populate_instruction(&mut instruction_data, 43, 2, 1);
+
+    assert_eq!(
+        process_instruction(
+            &program_id,
+            &[
+                funding_account.clone(),
+                price_account.clone(),
+                clock_account.clone()
+            ],
+            &instruction_data
+        ),
+        Err(ProgramError::InvalidArgument)
+    );
+
+    {
+        let price_data = load_checked::<PriceAccount>(&price_account, PC_VERSION).unwrap();
+        assert_eq!(price_data.comp_[0].latest_.price_, 42);
+        assert_eq!(price_data.comp_[0].latest_.conf_, 2);
+        assert_eq!(price_data.comp_[0].latest_.pub_slot_, 1);
+        assert_eq!(price_data.comp_[0].latest_.status_, PC_STATUS_TRADING);
+        assert_eq!(price_data.valid_slot_, 0);
+        assert_eq!(price_data.agg_.pub_slot_, 0);
+        assert_eq!(price_data.agg_.price_, 0);
+        assert_eq!(price_data.agg_.status_, PC_STATUS_UNKNOWN);
+    }
+
+    // We aggregate the price at the end of each slot now.
+    validator::aggregate_price(1, 101, price_account.key, *price_account.data.borrow_mut())
+        .unwrap();
+    update_clock_slot(&mut clock_account, 2);
+
+    validator::aggregate_price(2, 102, price_account.key, *price_account.data.borrow_mut())
+        .unwrap();
```

Comment:
nit: can we once verify the output of this as well?

Reply:
Added some output checks.